bigquery/src/main/java/org/apache/zeppelin/bigquery/bigQueryInterpreter.java

```diff
@@ -0,0 +1,340 @@
+    try {
+      for (TableFieldSchema schem: response.getSchema().getFields()) {
+        msg.append(schem.getName());
+        msg.append(TAB);
+      }      
+      msg.append(NEWLINE);
+      for (TableRow row : response.getRows()) {
+        for (TableCell field : row.getF()) {
+          msg.append(field.getV().toString());
+          msg.append(TAB);
+        }
+        msg.append(NEWLINE);
+      }
+      return msg.toString();
+    }
+    catch ( NullPointerException ex ) {
+      throw new NullPointerException("SQL Execution returned an error!");
+    }
+  }
+
+  //Function to poll a job for completion. Future use
+  public static Job pollJob(final Bigquery.Jobs.Get request, final long interval) 
+      throws IOException, InterruptedException {
+    Job job = request.execute();
+    while (!job.getStatus().getState().equals("DONE")) {
+      System.out.println("Job is " 
+          + job.getStatus().getState() 
+          + " waiting " + interval + " milliseconds...");
+      Thread.sleep(interval);
+      job = request.execute();
+    }
+    return job;
+  }
+
+  //Function to page through the results of an arbitrary bigQuery request
+  public static <T extends GenericJson> Iterator<T> getPages(
+      final BigqueryRequest<T> requestTemplate) {
+    class PageIterator implements Iterator<T> {
+      private BigqueryRequest<T> request;
+      private boolean hasNext = true;
+      public PageIterator(final BigqueryRequest<T> requestTemplate) {
+        this.request = requestTemplate;
+      }
+      public boolean hasNext() {
+        return hasNext;
+      }
+      public T next() {
+        if (!hasNext) {
+          throw new NoSuchElementException();
+        }
+        try {
+          T response = request.execute();
+          if (response.containsKey("pageToken")) {
+            request = request.set("pageToken", response.get("pageToken"));
+          } 
+          else {
+            hasNext = false;
+          }
+          return response;
+        } 
+        catch (IOException e) {
+          e.printStackTrace();
+          return null;
+        }
+      }
+
+      public void remove() {
+        this.next();
+      }
+    }
+
+    return new PageIterator(requestTemplate);
+  }
+  
+  //Function to call bigQuery to run SQL and return results to the Interpreter for output
+  private InterpreterResult executeSql(String sql) {
+    int counter = 0;
+    StringBuilder finalmessage = null;
+    finalmessage = new StringBuilder("%table ");
+    String projId = getProperty(PROJECT_ID);
+    long wTime = Long.parseLong(getProperty(WAIT_TIME));
+    long maxRows = Long.parseLong(getProperty(MAX_ROWS));
+    Iterator<GetQueryResultsResponse> pages = run(sql, projId, wTime, maxRows);
+    try {
+      while (pages.hasNext()) {
+        finalmessage.append(printRows(pages.next()));
+      }
+      return new InterpreterResult(Code.SUCCESS, finalmessage.toString());
+    }
+    catch ( NullPointerException ex ) {
```

Comment:
a nit: looks like indentation is a bit off here.
To be consistent with the rest of the project and [styleguide](https://google.github.io/styleguide/javaguide.html#s6.2-caught-exceptions) here and in other places `catch` needs to be on prev. line